### PR TITLE
Add template endpoint.

### PR DESCRIPTION
**Pull request by enozcan**
I've added a template endpoint. Try not to test it since it will not work. Its purpose is to show which project structure has to be followed when creating a new endpoint.

### CODE REVIEW

File: backend/TraderX/src/main/java/cmpe451/group6/rest/template/model/TemplateItem.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+/**
+ *  Here, you create an object which is
+ *  going to be stored in database directly.
+ *
+ *  [DO NOT CHANGE OR IMPLEMENT THOSE CLASSES / INTERFACES. FOR TEMPLATING PURPOSE ONLY.]
+ */
+@Entity(name = "OptionalTableName")
+public class TemplateItem {
+
```
Comment (barisegesevgili) [inline backend/TraderX/src/main/java/cmpe451/group6/rest/template/model/TemplateItem.java]: adding @Data label to the beginning of that class is automatically creates getter and setter methods AFAIK

Comment (enozcan) [reply backend/TraderX/src/main/java/cmpe451/group6/rest/template/model/TemplateItem.java]: IMO, creating getters & setters is not a big deal thanks to the IDE shortcuts. 
Hence that's not that necessary when we already have dozens of annotations like SwaggerAPI etc.

**Comment by burakikanyildiz** [comment]
I have checked your changes. It looks good to me and it works properly

**Comment by enozcan** [comment]
@barisegesevgili 
Thanks for the password regex. LGTM. I thinks it's OK to be merged now.

**Pull request merged**